src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs

```diff
@@ -1098,18 +1098,18 @@ private void CheckInputColumnTypes(DataViewSchema schema)
 
             var t = schema[(int)LabelIndex].Type;
             if (t != NumberDataViewType.Single && t != NumberDataViewType.Double && t != BooleanDataViewType.Instance && t.GetKeyCount() != 2)
-                throw Host.ExceptSchemaMismatch(nameof(schema), "label", LabelCol, "float, double, bool or a KeyType with cardinality 2", t.ToString());
```

Comment:
>"float, double, bool  [](start = 83, length = 21)

i'd live them

Reply:
[https://docs.microsoft.com/en-us/dotnet/standard/language-independence-and-language-independent-components?view=netframework-4.7.2#types-and-type-member-signatures](https://docs.microsoft.com/en-us/dotnet/standard/language-independence-and-language-independent-components?view=netframework-4.7.2#types-and-type-member-signatures)
If you scroll down it has list of CLS compliant types.
float and double are C# specific, while Single and Double I believe should work on vbasic, F# and other languages.


---
In reply to: [268223604](https://github.com/dotnet/machinelearning/pull/3046#discussion_r268223604) [](ancestors = 268223604)